server/src/main/java/org/opensearch/action/main/TransportMainAction.java

```diff
@@ -33,36 +33,51 @@
 package org.opensearch.action.main;
 
 import org.opensearch.Build;
+import org.opensearch.LegacyESVersion;
 import org.opensearch.Version;
 import org.opensearch.action.ActionListener;
 import org.opensearch.action.support.ActionFilters;
 import org.opensearch.action.support.HandledTransportAction;
 import org.opensearch.cluster.ClusterState;
 import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Setting;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.node.Node;
 import org.opensearch.tasks.Task;
 import org.opensearch.transport.TransportService;
 
 public class TransportMainAction extends HandledTransportAction<MainRequest, MainResponse> {
-
     private final String nodeName;
     private final ClusterService clusterService;
+    private final String responseVersionNumber;
+
+    // HTTP_MAIN_RESPONSE_VERSION_NUMBER exists to provide support for clients expecting legacy ES versions
+    // to be returned from the main rest endpoint.  This setting will be removed in future versions of OpenSearch.
+    public static final Setting<String> SETTING_HTTP_MAIN_RESPONSE_VERSION_NUMBER =
+        Setting.simpleString("http.main.response.version", Build.CURRENT.getQualifiedVersion(),
+            value -> {
+                if (value.equals(Build.CURRENT.getQualifiedVersion()) == false &&
+                    LegacyESVersion.fromString(value).onOrBefore(LegacyESVersion.V_7_10_2) == false) {
```

Comment:
It should just be 7.10.2. what is the use case to give option to choose any version < 7.10.2. What if someone set 6.8. The code base will not automatically switch to 6.8 logic, this will add more confusion.

Reply:
Hardcoding this to 7.10.2 makes the assumption that the client is upgrading from 7.10.2.  We want to allow them to upgrade without forcing an upgrade to their client.

Reply:
What happens if I set the version to 6.8.0 and some client now believes I have a 6.x REST API? What was removed between 6 and 7?

Reply:
Good point, this is too much of a risk.  I don't think the original intent here of the setting is realistic and this should instead be a boolean to set to 7.10.2 directly similar to what is happening [here](https://github.com/opensearch-project/OpenSearch/blob/c62884a1e91606c3ddf663bf3b054148c6c493ec/server/src/main/java/org/opensearch/action/main/MainResponse.java#L106).